### ISSUE TITLE
 'bool' object has no attribute 'to_deterministic'

### ISSUE THREAD

**Issue by AloshkaD** [matterport/Mask_RCNN#723]
@waleedka Setting augmentation to True in `load_image_gt` Gives the error  'bool' object has no attribute 'to_deterministic'
That's caused by 
`det = augmentation.to_deterministic()`
Why would you want to convert a boolian ?

**Comment by keineahnung2345** [comment]
Just as load_image_gt() says, 

> augmentation: Optional. An imgaug (https://github.com/aleju/imgaug) augmentation.

one should pass a imgaug augmentation object rather than a boolean object into this argument.
Here's the example:
`augmentation = imgaug.augmenters.Fliplr(0.5)`

**Comment by javierfs** [comment]
what should we change, then? @AloshkaD @keineahnung2345

**Comment by keineahnung2345** [comment]
@javierfs If you don't want to do image augmentation, just pass `augmentation=None`. Otherwise you should pass something like `imgaug.augmenters.Fliplr(0.5)` to `data_generator`.

**Comment by HAMZARaouia** [comment]
Did you manage to solve it @AloshkaD  ? I have the same error now

**Comment by thanawat-hub** [comment]
hi everyone @AloshkaD  @javierfs @HAMZARaouia 

 if file model.py in mrcnn have something like this
 def train(self, train_dataset, val_dataset, learning_rate, epochs, layers,
              augmentation=True, custom_callbacks=None, no_augmentation_sources=None):
then this train() will passing augmentation = True to 
data_generator(train_dataset, self.config, shuffle=True,
                                         augmentation=augmentation,
                                         batch_size=self.config.BATCH_SIZE,
                                         no_augmentation_sources=no_augmentation_sources)
and data_generator(dataset, config, shuffle=True, augment=False, augmentation=True,
                   random_rois=0, batch_size=1, detection_targets=False,
                   no_augmentation_sources=None):
will passing aumgentation = True to 
 load_image_gt()
then  det = augmentation.to_deterministic() 
is will be error because boolean (True) have no function to_deterministic()

then like he said you can change in main.py (file train) to 

augmentation = imgaug.augmenters.Fliplr(0.5) 
or 
augmentation = iaa.SomeOf((0, 2), [
       iaa.Fliplr(0.5),
       iaa.Flipud(0.5),
       iaa.OneOf([iaa.Affine(rotate=90),
                  iaa.Affine(rotate=180),
                  iaa.Affine(rotate=270)]),
       iaa.Multiply((0.8, 1.5)),
       iaa.GaussianBlur(sigma=(0.0, 5.0))
    ])

augmentation type is imgaug.augmenters.meta can look in https://imgaug.readthedocs.io/en/latest/source/api_augmenters_meta.html
augmentation can.to_deterministic()

then change in train(), data_generator(),  and  load_image_gt() in model.py
like this 
 def train(self, train_dataset, val_dataset, learning_rate, epochs, layers,
              augmentation, custom_callbacks=None, no_augmentation_sources=None):

def load_image_gt(dataset, config, image_id, augment=False, augmentation,
                  use_mini_mask=False):

def data_generator(dataset, config, shuffle=True, augment=False, augmentation,
                   random_rois=0, batch_size=1, detection_targets=False,
                   no_augmentation_sources=None):
then run 
!python3 main.py --dataset=dataset --weights=coco
Im using colab
Hope This Help!!

**Comment by jacksonwu09** [comment]
@keineahnung2345 Do you solve it?
